### DISCUSSION TITLE
Building failed on Ubuntu

### DISCUSSION THREAD

**Discussion by August1996** [google/filament#5337] [Q&A] [answered]
# Build Common
```
CC=/usr/bin/clang-7 CXX=/usr/bin/clang++-7 CXXFLAGS='-stdlib=libc++ -fno-builtin -Wno-pass-failed' \
    ./build.sh  -p android -i release
```

# Error Message
```
Execution failed for task ':filament-android:configureCMakeRelWithDebInfo[arm64-v8a]'.
> [CXX1410] /root/filament-build/filament/android/filament-android/.cxx/RelWithDebInfo/1l6k252v/arm64-v8a/android_gradle_build.json release|arm64-v8a : expected buildTargetsCommandComponents command 'ninja' to exist
```

# Environment
```
jdk: openjdk 11.0.14 2022-01-18
ninja: 1.10.0
cmake: 3.19.0
```

**Comment by romainguy** [comment, marked as answer]
Looks like an issue with your Android SDK or NDK maybe. That error message comes from the Gradle build. Is your ninja correctly in PATH?

**Reply by August1996** [reply]
> Looks like an issue with your Android SDK or NDK maybe. That error message comes from the Gradle build. Is your ninja correctly in PATH?

i use the command(apt-get install ninja) to install ninja, and how can i set the path of ninja?

**Reply by August1996** [reply]
thanks, i try to use `./gradlew gltfio-android:assembleRelease` to build directly, it works.

the work directory is `filament/android`

**Reply by grassydragon** [reply]
It is a known bug in the build system that causes Gradle to search for ninja in the directory where CMake is installed: https://issuetracker.google.com/issues/206099937#comment1
The suggested workaround is to create a symlink to ninja in that directory.
